Tests/Services/TasksServiceTests.cs

```diff
@@ -415,4 +408,412 @@ await taskService.SubmitTasksAsync(new SessionInfo("sessionId1"),
+                              Filter        = new Filters(),
+                            };
+
+    var result = await taskService.ListTasksAsync(paginationOptions)
+                                  .ToListAsync();
+
+    ClassicAssert.IsNotNull(result);
+    ClassicAssert.AreEqual(1,
+                           result.Count);
+    ClassicAssert.AreEqual(2,
+                           result[0].TotalTasks);
+
+    var tasksData = result[0]
+                    .TasksData.ToList();
+    ClassicAssert.AreEqual(2,
+                           tasksData.Count);
+
+    var expectedTaskIds = new List<string>
+                          {
+                            "taskId1",
+                            "taskId2",
+                          };
+    var expectedStatuses = new List<TaskStatus>
+                           {
+                             TaskStatus.Completed,
+                             TaskStatus.Cancelling,
+                           };
+
+    for (var i = 0; i < tasksData.Count; i++)
+    {
+      ClassicAssert.AreEqual(expectedTaskIds[i],
+                             tasksData[i].Item1);
+      ClassicAssert.AreEqual(expectedStatuses[i],
+                             tasksData[i].Item2);
+    }
+  }
+
+  [Test]
+  public async Task GetTaskDetailedAsyncShouldReturnCorrectTaskDetails()
+  {
+    var taskResponse = new GetTaskResponse
+                       {
+                         Task = new TaskDetailed
+                                {
+                                  Id = "taskId1",
+                                  ExpectedOutputIds =
+                                  {
+                                    "outputId1",
+                                  },
+                                  DataDependencies =
+                                  {
+                                    "dependencyId1",
+                                  },
+                                  CreatedAt = Timestamp.FromDateTime(DateTime.UtcNow),
+                                  StartedAt = Timestamp.FromDateTime(DateTime.UtcNow.AddMinutes(-5)),
+                                  EndedAt   = Timestamp.FromDateTime(DateTime.UtcNow.AddMinutes(-1)),
+                                  SessionId = "sessionId1",
+                                },
+                       };
+
+    var mockInvoker = new Mock<CallInvoker>();
+    var callInvoker = mockInvoker.SetupAsyncUnaryCallInvokerMock<GetTaskRequest, GetTaskResponse>(taskResponse);
+
+    var taskService = MockHelper.GetTasksServiceMock(callInvoker);
+
+    var result = await taskService.GetTasksDetailedAsync("taskId1");
+
+    ClassicAssert.IsNotNull(result);
+    ClassicAssert.AreEqual("taskId1",
+                           result.TaskId);
+    ClassicAssert.AreEqual("outputId1",
+                           result.ExpectedOutputs.First());
+    ClassicAssert.AreEqual("dependencyId1",
+                           result.DataDependencies.First());
```

Comment:
Please use the [Contraint model](https://docs.nunit.org/articles/nunit/writing-tests/assertions/assertions.html) of assertions.